src/storage/src/table/cell_based_table.rs

```diff
@@ -456,3 +466,66 @@ impl<S: StateStore> TableIter for CellBasedTableRowIter<S> {
         }
     }
 }
+
+/// [`CellBasedTableStreamingIter`] is used for streaming executor, and will not wait for epoch.
+pub struct CellBasedTableStreamingIter<S: StateStore> {
+    /// An iterator that returns raw bytes from storage.
+    iter: StripPrefixIterator<S::Iter>,
+    /// Cell-based row deserializer
+    cell_based_row_deserializer: CellBasedRowDeserializer,
+    pub cell_based_item: Option<(Vec<u8>, Row)>,
+}
+
+impl<S: StateStore> CellBasedTableStreamingIter<S> {
+    pub async fn new(
+        keyspace: &Keyspace<S>,
+        table_descs: &[ColumnDesc],
+        epoch: u64,
+    ) -> StorageResult<Self> {
+        let cell_based_row_deserializer = CellBasedRowDeserializer::new(table_descs.to_vec());
+        let iter = keyspace.iter(epoch).await?;
+        let iter = Self {
+            iter,
+            cell_based_row_deserializer,
+            cell_based_item: None,
+        };
+        Ok(iter)
+    }
+
+    pub async fn peek(&mut self) -> StorageResult<Option<&(Vec<u8>, Row)>> {
+        if self.cell_based_item.is_none() {
+            self.cell_based_item = self.next().await?;
+        }
+        Ok(self.cell_based_item.as_ref())
+    }
+
+    /// return a row with its pk.
+    pub async fn next(&mut self) -> StorageResult<Option<(Vec<u8>, Row)>> {
+        loop {
+            match self.iter.next().await? {
+                None => {
+                    let pk_and_row = self.cell_based_row_deserializer.take();
+                    self.cell_based_item = pk_and_row.clone();
+                    return Ok(pk_and_row);
+                }
+                Some((key, value)) => {
+                    tracing::trace!(
```

Comment:
should this trace be removed? https://github.com/singularity-data/risingwave/issues/1421

Reply:
We sometimes need this trace for debugging reasons. Also, trace targets beginning with `events::` won't be enabled by default.

Reply:
Maybe these should be configured as `debug!` instead?

The perf impact of tracing can be quite high, especially for tokio tracing

Can see the work done in minitrace:
https://www.youtube.com/watch?v=Z5DXvhh3GVg&ab_channel=TiKV
https://github.com/tikv/minitrace-rust

Reply:
per-row info for ordinary jaeger trace seems too verbose to my mind

Reply:
If tokio-tracing is enabled, `debug!` will also be piped to tracing subscribers 🤣

Also we disabled log level < info in release mode. So this line won't be compiled in release mode indeed.

Reply:
By the way, minitrace doesn't provide logging functionality, it is a pure time-span tracing library. We have had plans to use minitrace for all time spans, but it is not being actively worked on.